aten/src/ATen/native/Unique.cpp

```diff
@@ -0,0 +1,58 @@
+// Returns unique elements of input tensor.
+
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+
+#include <set>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace at {
+namespace native{
+
+namespace {
+
+template <template <class...> class set_type, typename scalar_t>
+std::tuple<Tensor, Tensor> _unique_cpu_template(
+    const Tensor& self,
+    const bool return_inverse) {
+  const Tensor& input = self.contiguous();
+  set_type<scalar_t> set(
+      input.data<scalar_t>(), input.data<scalar_t>() + input.numel());
+  Tensor output = input.type().tensor({static_cast<long long>(set.size())});
+  std::copy(set.begin(), set.end(), output.data<scalar_t>());
+
+  Tensor inverse_indices = self.type().toScalarType(kLong).tensor({0});
+  if (return_inverse) {
+    inverse_indices.resize_(input.sizes());
+    std::unordered_map<scalar_t, int64_t> inverse_map;
+    inverse_map.reserve(output.numel());
+    for (int i = 0; i < output.numel(); ++i) {
+      inverse_map[output.data<scalar_t>()[i]] = i;
+    }
+    for (int i = 0; i < input.numel(); ++i) {
+      inverse_indices.data<int64_t>()[i] =
+          inverse_map[input.data<scalar_t>()[i]];
```

Comment:
I think you forgot to resize the tensor, so this will result in invalid memory accesses. Also, take the `.data` outside of the loop, it has a bunch of type checks inside IIRC and we don't want to do them at every step

Reply:
Some lines above, I called `inverse_indices.resize_(input.sizes());`? Or are you referring to something else?

Done - refactored out `.data` calls.